tests/pthread_barrier/tests/01-run.py

```diff
@@ -16,4 +16,4 @@ def testfunc(child):
 
 
 if __name__ == "__main__":
-    sys.exit(run(testfunc))
+    sys.exit(run(testfunc, timeout=30))
```

Comment:
I added this for `arduino-mega2560` that timeouts when using `test_utils_interactive_sync`, since the test doesn't immediately start. I'll improove the commit or splir.

Reply:
Will fix in a different way and split.

Reply:
addressed